### PR TITLE
translate timeRemaining()

### PR DESCRIPTION
**Pull request by leenty**

### CODE REVIEW

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md
Comment (fwqaaq) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:1]: 可以查看完整的翻译指南：<https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md>

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 1]
```diff
@@ -5,35 +5,36 @@ slug: Web/API/IdleDeadline/timeRemaining
 
 {{APIRef("Background Tasks")}}{{SeeCompatTable}}
 
-The **`timeRemaining()`** method on the {{domxref("IdleDeadline")}} interface returns the estimated number of milliseconds remaining in the current idle period. The callback can call this method at any time to determine how much time it can continue to work before it must return. For example, if the callback finishes a task and has another one to begin, it can call `timeRemaining()` to see if there's enough time to complete the next task. If there isn't, the callback can just return immediately, or look for other work to do with the remaining time.
+{{domxref("IdleDeadline")}} 接口上的 **`timeRemaining()`** 方法返回当前空闲期间剩余的估计毫秒数。回调函数可以在任何时候调用此方法，以确定在必须返回之前它可以继续工作多长时间。例如，如果回调完成了一个任务，并且有另一个任务要开始，它可以调用 `timeRemaining()` 来查看是否有足够的时间来完成下一个任务。如果没有，回调函数可以立即返回，或者用剩余的时间寻找其他合适的工作。
 
-By the time `timeRemaining()` reaches 0, it is suggested that the callback should return control to the user agent's event loop.
+当 `timeRemaining()` 达到0时，建议回调应该将控制权返回给用户代理的事件循环。
```
Comment (fwqaaq) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:10]:
```suggestion
当 `timeRemaining()` 达到 0 时，建议回调应该将控制权返回给用户代理的事件循环。
```

see <https://github.com/mdn/translated-content/blob/main/docs/zh-cn/translation-guide.md>

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 2]
```diff
@@ -5,35 +5,36 @@ slug: Web/API/IdleDeadline/timeRemaining
...
 
 {{Specifications}}
 
-## Browser compatibility
+## 浏览器兼容性
 
 {{Compat}}
 
-## See also
+## 相关链接
```
Comment (fwqaaq) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:37]:
```suggestion
## 参见
```

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 3]
```diff
@@ -5,35 +5,36 @@ slug: Web/API/IdleDeadline/timeRemaining
...
-The **`timeRemaining()`** method on the {{domxref("IdleDeadline")}} interface returns the estimated number of milliseconds remaining in the current idle period. The callback can call this method at any time to determine how much time it can continue to work before it must return. For example, if the callback finishes a task and has another one to begin, it can call `timeRemaining()` to see if there's enough time to complete the next task. If there isn't, the callback can just return immediately, or look for other work to do with the remaining time.
+{{domxref("IdleDeadline")}} 接口上的 **`timeRemaining()`** 方法返回当前空闲期间剩余的估计毫秒数。回调函数可以在任何时候调用此方法，以确定在必须返回之前它可以继续工作多长时间。例如，如果回调完成了一个任务，并且有另一个任务要开始，它可以调用 `timeRemaining()` 来查看是否有足够的时间来完成下一个任务。如果没有，回调函数可以立即返回，或者用剩余的时间寻找其他合适的工作。
 
-By the time `timeRemaining()` reaches 0, it is suggested that the callback should return control to the user agent's event loop.
+当 `timeRemaining()` 达到 0 时，建议回调应该将控制权返回给用户代理的事件循环。
 
-## Syntax
+## 语法
 
 ```plain
```
Comment (yin1999) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:14]:
````suggestion
```js-nolint
````

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 4]
```diff
@@ -5,35 +5,36 @@ slug: Web/API/IdleDeadline/timeRemaining
...
 timeRemaining = IdleDeadline.timeRemaining();
 ```
 
-### Return value
+### 返回值
 
-A {{domxref("DOMHighResTimeStamp")}} value (which is a floating-point number) representing the number of milliseconds the user agent estimates are left in the current idle period. The value is ideally accurate to within about 5 microseconds.
+一个 {{domxref("DOMHighResTimeStamp")}} 值(是一个浮点数)，表示用户代理估计在当前空闲期间内剩余的毫秒数。理想情况下，该值精确到大约5微秒。
```
Comment (yin1999) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:20]:
```suggestion
一个 {{domxref("DOMHighResTimeStamp")}} 值（是一个浮点数），表示用户代理估计在当前空闲期间内剩余的毫秒数。理想情况下，该值精确到大约 5 微秒。
```

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 5]
```diff
@@ -5,35 +5,36 @@ slug: Web/API/IdleDeadline/timeRemaining
...
 
-If the {{domxref("IdleDeadline")}} object's {{domxref("IdleDeadline.didTimeout", "didTimeout")}} property is true, this method returns zero.
+如果 {{domxref("IdleDeadline")}} 对象的 {{domxref("IdleDeadline.didTimeout", "didTimeout")}} 属性为 `true` ，该方法返回 `0`。
 
-## Example
+## 示例
 
 See our [complete example](/zh-CN/docs/Web/API/Background_Tasks_API#Example) in the article [Cooperative Scheduling of Background Tasks API](/zh-CN/docs/Web/API/Background_Tasks_API).
```
Comment (yin1999) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:26]:
```suggestion
```

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 6]
```diff
@@ -5,35 +5,36 @@ slug: Web/API/IdleDeadline/timeRemaining
...
+在文章[Cooperative Scheduling of Background Tasks API](/zh-CN/docs/Web/API/Background_Tasks_API) 中可以查看 [完整示例](/zh-CN/docs/Web/API/Background_Tasks_API#Example) 。
```
Comment (yin1999) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:27]:
```suggestion
在文章[后台任务协作调度 API](/zh-CN/docs/Web/API/Background_Tasks_API) 中可以查看[完整示例](/zh-CN/docs/Web/API/Background_Tasks_API#示例) 。
```

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 7]
```diff
@@ -5,35 +5,36 @@ slug: Web/API/IdleDeadline/timeRemaining
...
 
-## Browser compatibility
+## 浏览器兼容性
 
 {{Compat}}
 
-## See also
+## 参见
 
 - [Collaborative Scheduling of Background Tasks](/zh-CN/docs/Web/API/Background_Tasks_API)
```
Comment (yin1999) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:39]:
```suggestion
- [后台任务协作调度 API](/zh-CN/docs/Web/API/Background_Tasks_API)
```

**Review by yin1999 — APPROVED** [review]

File: files/zh-cn/web/api/idledeadline/timeremaining/index.md [hunk 8]
```diff
@@ -5,37 +5,37 @@ slug: Web/API/IdleDeadline/timeRemaining
 
 {{APIRef("Background Tasks")}}{{SeeCompatTable}}
```
Comment (yin1999) [inline files/zh-cn/web/api/idledeadline/timeremaining/index.md:6]:
```suggestion
{{APIRef("Background Tasks")}}
```

**Comment by yin1999** [comment]
Thanks @leenty. Welcome aboard :tada:

**Pull request merged**